src/V3EmitC.cpp

```diff
@@ -1337,16 +1377,85 @@ class EmitVarTspSorter final : public V3TSP::TspStateBase {
 
 unsigned EmitVarTspSorter::s_serialNext = 0;
 
+//######################################################################
+// Emit lazy forward declarations
+
+class LazyDecls final : public AstNVisitor {
+    // NODE STATE/TYPES
+    //  AstNode::user2() -> bool. Already emitted decl for symbols.
+    AstUser2InUse m_inuser2;
+
+    // MEMBERS
+    std::unordered_set<string> m_emittedManually;  // Set of names already declared manually.
+    EmitCBaseVisitor& m_emitter;  // For access to file output
+    bool m_needsBlankLine;  // Emit blank line if any declarations are emitted (purely cosmetic)
```

Comment:
```suggestion
    bool m_needsBlankLine = false;  // Emit blank line if any declarations are emitted (purely cosmetic)
```

Reply:
This would be redudant, as it's reset in the `emit` function below where this member is read. Could rewite that to reset to false on exit rather than entry if you insist.

Reply:
Any variable not reset here or in the constructor will cause some lint tools to complain.  I see your point about needing it in emit, but see no harm in also defaulting it, just good defense.

Reply:
Ok, lint tools are fair point. I think I have spent too much time trying to strip redundancy from the generated code to be comfortable with these kind of things ;)